metabase/metabase
# Add syncing state snackbar & sample dataset modal

Epic: https://github.com/metabase/metabase/issues/18675

Adds a Snackbar to show the progress of the initial database sync:
<img width="274" alt="Screen Shot 2021-11-05 at 6 59 42 pm" src="https://user-images.githubusercontent.com/8542534/140540887-dae5206f-1e38-4ae9-8680-861593f0cce5.png">

Adds a modal to navigate the user to Sample Dataset:
<img width="672" alt="Screen Shot 2021-11-05 at 6 59 38 pm" src="https://user-images.githubusercontent.com/8542534/140540883-0cb2d837-c054-4dbf-bd0e-e2d4a857b852.png">

**How to test**
As the BE for this PR is not ready we would need to rely on mocked data:
1. Set `initial_sync` to false https://github.com/metabase/metabase/blob/c0b5c697d1c5c814ec35a50ed2696617ea172d4a/frontend/src/metabase/schema.js#L19
2. Set `initial_sync` to false https://github.com/metabase/metabase/blob/c0b5c697d1c5c814ec35a50ed2696617ea172d4a/frontend/src/metabase/schema.js#L53
3. Set the setting to true (return true) https://github.com/metabase/metabase/blob/52222d2e86757a913eea3c5e54d7a13e0694f648/frontend/src/metabase/syncing/selectors.js#L47

Setup steps:
1.  Add a database during the setup. You can copy the connection string from the Sample Dataset and add a new H2 database.
2. You should be able to see the modal and the snackbar when redirected to Home page.
3. Snackbar should poll the BE for new data about the tables.

Admin steps:
1. Add a new database. You can copy the connection string from the Sample Dataset and add a new H2 database.
2. You should be able to see the modal and the snackbar.
3. Snackbar should poll the BE for new data about the tables.

Please note that it should be OK to merge this PR before BE changes as with `initial_sync` set statically to `true` users won't see FE changes.
